src/main/java/org/javacs/InferConfig.java

```diff
@@ -22,20 +22,33 @@
     private final Path mavenHome;
     /** Location of the gradle cache, usually ~/.gradle */
     private final Path gradleHome;
+    /** VSCode's current working directory */
+    private final String target;
 
-    InferConfig(Path workspaceRoot, Collection<String> externalDependencies, Path mavenHome, Path gradleHome) {
+    InferConfig(
```

Comment:
Why is workspaceroot different than cwd? My understanding of how VSCode works is they should always be the same.

Reply:
I've been using this with vim + ale. We have a large monorepo, so my workflow often involves working in a subdirectory of the workspace root. I haven't tested with VSCode. I assume you'd be able to use VSCode in a similar manner though.

Reply:
As far as I can tell, the cwd that the language server process runs in isn't part of the LSP, so we shouldn't rely on it for anything. When you're working in a subdirectory of a large monorepo, workspaceRoot should be the subdirectory, not the parent directory.

Reply:
Does the code currently walk up from the workspaceRoot looking for the bazel WORKSPACE file? I think if we do that, then leaving the workspaceRoot == cwd will still work for me and allow me to add the relative path from the bazel WORKSPACE -> workspaceRoot in the bazel commands. Does that make sense?

Reply:
I think that's what it does. If it doesn't, it should.

Reply:
It doesn't seem to recursively search parent directories for the bazel `WORKSPACE` file. It looks like it checks the `workspaceRoot` only. At least, that's all it appeared to do for me.